ts/features/itwallet/common/components/ItwDigitalVersionBadge.tsx

```diff
@@ -5,43 +5,56 @@ import {
   IOColors,
   makeFontStyleObject
 } from "@pagopa/io-app-design-system";
-import React from "react";
+import Color from "color";
+import React, { useMemo } from "react";
 import { Platform, StyleSheet, Text, View } from "react-native";
 import I18n from "../../../../i18n";
 
+export type TagColorScheme = "default" | "faded" | "greyscale";
+
 type DigitalVersionBadgeProps = {
   credentialType: string;
-  isFaded?: boolean;
+  colorScheme: TagColorScheme;
 };
 
 type CredentialTypesProps = {
   background: string;
   foreground: string;
 };
 
+const mapCredentialTypes: Record<string, CredentialTypesProps> = {
+  MDL: {
+    foreground: "#5E303E",
+    background: "#FADCF5"
+  },
+  EuropeanDisabilityCard: {
+    foreground: "#01527F",
+    background: "#E8EEF4"
+  },
+  EuropeanHealthInsuranceCard: {
+    foreground: "#032D5C",
+    background: "#ABD8F2"
+  }
+};
+
 const ItwDigitalVersionBadge = ({
   credentialType,
-  isFaded = false
+  colorScheme = "default"
 }: DigitalVersionBadgeProps) => {
-  const mapCredentialTypes: Record<
-    NonNullable<string>,
-    CredentialTypesProps
-  > = {
-    MDL: {
-      foreground: "#5E303E",
-      background: "#FADCF5"
-    },
-    EuropeanDisabilityCard: {
-      foreground: "#01527F",
-      background: "#E8EEF4"
-    },
-    EuropeanHealthInsuranceCard: {
-      foreground: "#032D5C",
-      background: "#ABD8F2"
+  const colorProps = useMemo(() => {
+    const baseColorProps = mapCredentialTypes[credentialType];
+    if (!baseColorProps) {
+      return;
+    }
+    if (colorScheme === "greyscale") {
+      return {
+        foreground: Color(baseColorProps.foreground).grayscale().hex(),
+        background: Color(baseColorProps.background).grayscale().hex()
+      };
     }
-  };
+    return baseColorProps;
+  }, [credentialType, colorScheme]);
```

Comment:
I think that this `useMemo` is a bit redundant and does not provide any benefit. `colorProps` depends on the same props as the component, it will be recalculated every time the component re-renders anyway.
Keeping the memoization has negligible overhead, tho.
Just a note, feel free to do what you think is most appropriate!

Reply:
You are right, I have removed two `useMemo` that were dependent on all the component props (2f0a36e3e098396889eb939d4adf77dde7cc749c).